test/resources/cider-doc.edn

```diff
@@ -0,0 +1,2 @@
+{[cider-doc-edn-example] {:info-provider cider.nrepl.middleware.info-test/sample-info-provider
+                          :if            cider.nrepl.middleware.info-test/sample-info-provider?}}
```

Comment:
Besides from a function, `:if` could be a set of ns-qualified symbols:

```clj
:if #{clojure.core/future com.corp/my-macro ,,, }
```

which would mean: "this rule only applies if the symbol was found within one of these calls".

That would simplify user configs since, I reckon, most people won't want to do parsing themselves.